docs/src/tutorials/Nonlinear programs/space_shuttle_reentry_trajectory.jl

```diff
@@ -0,0 +1,312 @@
+# 
+# ```math
+# h = 80000 \text{ ft}, \qquad v = 2500 \text{ ft/sec}, \qquad \gamma = -5 \text{ deg}.
+# ```
+
+# As explained in the book, our goal is to maximize the final crossrange,
+# which is equivalent to maximizing the final latitude of the vehicle,
+# i.e., $J = \theta(t_F)$.
+
+# ![Max crossrange shuttle reentry](https://i.imgur.com/1tUN3wM.png)
+
+# ### Approach 
+# We will use a discretized model of time, with a fixed number of discretized points, $n$.
+# The decision variables at each point are going to be the state of the vehicle and the controls commanded to it.
+# In addition, we will also make each time step size $\Delta t$ a decision variable;
+# that way, we can either fix the time step size easily,
+# or allow the solver to fine-tune the duration between each adjacent pair of points.
+# Finally, in order to approximate the derivatives of the problem dynamics, we will use either rectangular
+# or [trapezoidal](http://en.wikipedia.org/wiki/Trapezoidal_rule) integration.
+
+# > #### Disclaimer
+# > Do not try to actually land a Space Shuttle using this notebook! 😛
+# > There's no mesh refinement going on, which can lead to unrealistic trajectories having position
+# > and velocity errors with orders of magnitude $10^4$ ft and $10^2$ ft/sec, respectively.
+
+using Interpolations
+using JuMP, Ipopt #, KNITRO
+
+## Global variables
+w  = 203000.0  # weight (lb)
+g₀ = 32.174    # acceleration (ft/sec^2)
+m  = w / g₀    # mass (slug)
+
+## Aerodynamic and atmospheric forces on the vehicle
+ρ₀ =  0.002378
+hᵣ =  23800.0
+Rₑ =  20902900.0
+μ  =  0.14076539e17
+S  =  2690.0
+a₀ = -0.20704
+a₁ =  0.029244
+b₀ =  0.07854
+b₁ = -0.61592e-2
+b₂ =  0.621408e-3
+c₀ =  1.0672181
+c₁ = -0.19213774e-1
+c₂ =  0.21286289e-3
+c₃ = -0.10117249e-5
+
+## Initial conditions
+hₛ = 2.6          # altitude (ft) / 1e5
+ϕₛ = deg2rad(0)   # longitude (rad)
+θₛ = deg2rad(0)   # latitude (rad)
+vₛ = 2.56         # velocity (ft/sec) / 1e4
+γₛ = deg2rad(-1)  # flight path angle (rad)
+ψₛ = deg2rad(90)  # azimuth (rad)
+αₛ = deg2rad(0)   # angle of attack (rad)
+βₛ = deg2rad(0)   # bank angle (rad)
+tₛ = 1.00         # time step (sec)
+
+## Final conditions, the so-called Terminal Area Energy Management (TAEM)
+hₜ = 0.8          # altitude (ft) / 1e5
+vₜ = 0.25         # velocity (ft/sec) / 1e4
+γₜ = deg2rad(-5)  # flight path angle (rad)
+
+## Number of mesh points (knots) to be used
+n = 2009
+
+## Integration scheme to be used for the dynamics
+integration_rule = "rectangular"
+nothing #hide
+
+#-
+
+user_options_ipopt = (
+    "mu_strategy" => "monotone",
+    "linear_solver" => "ma27",  # For the best results, it is advised to experiment different linear solvers.
+                                ## If Ipopt is not compiled with MA27/MA57, it may fallback to 'MUMPS'.
+                                ## In general, the linear solver MA27 is much faster than MUMPS.
```

Comment:
Does Ipopt not throw an error if MA27 is not available? This comment may be better structured as a `!!! tip`.
```
!!! tip
    For best results, it is advised ...
```